### PR TITLE
Created image gallery7

### PR DESCRIPTION
**Pull request by Avassaa**

### CODE REVIEW

**Review by akhnos — COMMENTED** [review]
using global css elements are forbidden, the component scss file can only contain classnames it reproduces

**Comment by Avassaa** [comment]
Tekrardan kontrol edebilirsiniz, kodu düzelttim

**Comment by dogusmiuw** [comment]
Only send image gallery 7 files.

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+    
+
+      getName(): string {
+        return "Image Gallery 7";
+      }
+    
+      render() {
+  const gallery = this.getPropValue("gallery") as CardItem[];
+  const repeatCount = this.getPropValue("cardRepeatTime") as number;
+  const cardBackgroundColor = this.getPropValue("cardBackgroundColor") as string || "#f1f1e8";
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:453]: I don't think this is going to work. Prop key is not correct.

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
Please handle the changes.

**Comment by dogusmiuw** [comment]
This is a new component. Import this component into registerables.tsx file.

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+            type: "array",
+            key: "gallery",
+            displayer: "Gallery",
+            value: 
+                [
+                    {
+
+                        type: "array",
+                        key: "card",
+                        displayer: "Card 1",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:41]: use singular and non-contain number names.
```suggestion
                        displayer: "Card",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+                        value: [
+                            { 
+                                type: "image",
+                                key: "image",
+                                displayer: "Image 1",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:46]: use singular and non-contain number names.
```suggestion
                        displayer: "Image",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/669dfff22f8a5b002ce60115?alt=media", 
+                            },
+                            {
+                                type: "string",
+                                key: "title",
+                                displayer: "title",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:52]:
```suggestion
                                displayer: "Title",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "Dubai",
+                            },
+                            {
+                                type: "string",
+                                key: "subtitle",
+                                displayer: "subtitle",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:58]:
```suggestion
                                displayer: "Subtitle",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+                                key:"backgroundColor",
+                                displayer:"Background Color",
+                                value:"#f1f1e8"
+                            }
+                        ]
+                    },
+                    {
+                        type: "array",
+                        key: "card",
+                        displayer: "Card 2",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:72]: use singular and non-contain number names.
```suggestion
                        displayer: "Card",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 7]
```diff
@@ -0,0 +1,490 @@
...
+                        value: [
+                            { 
+                                type: "image",
+                                key: "image",
+                                displayer: "Image 2",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:77]: use singular and non-contain number names.
```suggestion
                        displayer: "Image",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 8]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/669e00122f8a5b002ce60121?alt=media", 
+                            },
+                            {
+                                type: "string",
+                                key: "title",
+                                displayer: "title",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:83]:
```suggestion
                                displayer: "Title",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 9]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "Cosmoso",
+                            },
+                            {
+                                type: "string",
+                                key: "subtitle",
+                                displayer: "subtitle",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:89]:
```suggestion
                                displayer: "Subtitle",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 10]
```diff
@@ -0,0 +1,490 @@
...
+                                key:"backgroundColor",
+                                displayer:"Background Color",
+                                value:"#f1f1e8"
+                            }
+                        ]
+                    },
+                    {
+                        type: "array",
+                        key: "card",
+                        displayer: "Card 3",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:103]: use singular and non-contain number names.
```suggestion
                        displayer: "Card",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 11]
```diff
@@ -0,0 +1,490 @@
...
+                        value: [
+                            { 
+                                type: "image",
+                                key: "image",
+                                displayer: "Image 3",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:108]:
```suggestion
                                displayer: "Image",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 12]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/669e003a2f8a5b002ce6012d?alt=media", 
+                            },
+                            {
+                                type: "string",
+                                key: "title",
+                                displayer: "title",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:114]:
```suggestion
                                displayer: "Title",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 13]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "Ron Mcclenny",
+                            },
+                            {
+                                type: "string",
+                                key: "subtitle",
+                                displayer: "subtitle",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:120]:
```suggestion
                                displayer: "Subtitle",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 14]
```diff
@@ -0,0 +1,490 @@
...
+                                key:"backgroundColor",
+                                displayer:"Background Color",
+                                value:"#f1f1e8"
+                            }
+                        ]
+                    },
+                    {
+                        type: "array",
+                        key: "card",
+                        displayer: "Card 4",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:134]: use singular and non-contain number names.
```suggestion
                        displayer: "Card",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 15]
```diff
@@ -0,0 +1,490 @@
...
+                        value: [
+                            { 
+                                type: "image",
+                                key: "image",
+                                displayer: "Image 4",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:139]:
```suggestion
                                displayer: "Image",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 16]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/669e005b2f8a5b002ce60139?alt=media", 
+                            },
+                            {
+                                type: "string",
+                                key: "title",
+                                displayer: "title",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:145]:
```suggestion
                                displayer: "Title",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 17]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "Curitiba Brasil",
+                            },
+                            {
+                                type: "string",
+                                key: "subtitle",
+                                displayer: "subtitle",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:151]:
```suggestion
                                displayer: "Subtitle",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 18]
```diff
@@ -0,0 +1,490 @@
...
+                                key:"backgroundColor",
+                                displayer:"Background Color",
+                                value:"#f1f1e8"
+                            }
+                        ]
+                    }, 
+                    {
+                        type: "array",
+                        key: "card",
+                        displayer: "Card 5",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:165]: use singular and non-contain number names.
```suggestion
                        displayer: "Card",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 19]
```diff
@@ -0,0 +1,490 @@
...
+                        value: [
+                            { 
+                                type: "image",
+                                key: "image",
+                                displayer: "Image 5",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:170]:
```suggestion
                                displayer: "Image",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 20]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/669e00772f8a5b002ce60145?alt=media", 
+                            },
+                            {
+                                type: "string",
+                                key: "title",
+                                displayer: "title",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:176]:
```suggestion
                                displayer: "Title",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 21]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "John Doe",
+                            },
+                            {
+                                type: "string",
+                                key: "subtitle",
+                                displayer: "subtitle",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:182]:
```suggestion
                                displayer: "Subtitle",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 22]
```diff
@@ -0,0 +1,490 @@
...
+                                key:"backgroundColor",
+                                displayer:"Background Color",
+                                value:"#f1f1e8"
+                            }
+                        ]
+                    },
+                    {
+                        type: "array",
+                        key: "card",
+                        displayer: "Card 6",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:196]: use singular and non-contain number names.
```suggestion
                        displayer: "Card",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 23]
```diff
@@ -0,0 +1,490 @@
...
+                        value: [
+                            { 
+                                type: "image",
+                                key: "image",
+                                displayer: "Image 6",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:201]:
```suggestion
                                displayer: "Image",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 24]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/669e00952f8a5b002ce60151?alt=media", 
+                            },
+                            {
+                                type: "string",
+                                key: "title",
+                                displayer: "title",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:207]:
```suggestion
                                displayer: "Title",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 25]
```diff
@@ -0,0 +1,490 @@
...
+                                value: "Creme",
+                            },
+                            {
+                                type: "string",
+                                key: "subtitle",
+                                displayer: "subtitle",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:213]:
```suggestion
                                displayer: "Subtitle",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 26]
```diff
@@ -0,0 +1,490 @@
...
+                                displayer:"Background Color",
+                                value:"#f1f1e8"
+                            }
+                        ]
+                    }
+                    ,
+                    {
+                        type: "array",
+                        key: "card",
+                        displayer: "Card 7",
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:229]: use singular and non-contain number names. do this for all.
```suggestion
                        displayer: "Card",
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 27]
```diff
@@ -0,0 +1,490 @@
...
+  return (
+    <div
+      className={this.decorateCSS("imageGallery7")}
+      style={{
+        "--composer-card-background": cardBackgroundColor,
+      } as React.CSSProperties}
+    >
+      <div className={this.decorateCSS("content")}>
+        {Array.from({ length: 4 }).map((_, columnIndex: number) => (
+          <div key={columnIndex} className={`${this.decorateCSS("column")} ${this.decorateCSS(`column-${columnIndex}`)}`}>
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:464]: find a better way for this -> ```${this.decorateCSS(`column-${columnIndex}`)}```
```suggestion
          <div key={columnIndex} className={`${this.decorateCSS("column")} ${this.decorateCSS(`column-${columnIndex}`)}`}>
```

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 28]
```diff
@@ -0,0 +1,490 @@
...
+  const cardBackgroundColor = this.getPropValue("backgroundColor") as string || "#f1f1e8";
+ 
+  const repeatedGallery = Array.from({ length: repeatCount }).flatMap(() => gallery);
+
+  return (
+    <div
+      className={this.decorateCSS("imageGallery7")}
+      style={{
+        "--composer-card-background": cardBackgroundColor,
+      } as React.CSSProperties}
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:460]: doesn't seem right.

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 29]
```diff
@@ -0,0 +1,490 @@
...
+    >
+      <div className={this.decorateCSS("content")}>
+        {Array.from({ length: 4 }).map((_, columnIndex: number) => (
+          <div key={columnIndex} className={`${this.decorateCSS("column")} ${this.decorateCSS(`column-${columnIndex}`)}`}>
+            {repeatedGallery
+              .filter((_, index: number) => index % 4 === columnIndex)
+              .map((card, index) => {
+                const image = card.value.find(item => item.type === "image") as ImageType;
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:468]: dont cast values like this. use this.castToObject

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 30]
```diff
@@ -0,0 +1,490 @@
...
+                const title = card.value.find(item => item.key === "title") as unknown as { type: "string"; key: string; displayer: string; value: string };
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:469]: dont cast values like this. create a type for it and use this.castToObject.

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 31]
```diff
@@ -0,0 +1,490 @@
...
+                const subtitle = card.value.find(item => item.key === "subtitle") as unknown as  { type: "string"; key: string; displayer: string; value: string };
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:470]: dont cast values like this. create a type for it and use this.castToObject.

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 32]
```diff
@@ -0,0 +1,490 @@
...
+                const backgroundColor = card.value.find(item => item.key === "backgroundColor")?.value as string || cardBackgroundColor;
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:471]: dont cast values like this. create a type for it and use this.castToObject.

File: editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx [hunk 33]
```diff
@@ -0,0 +1,490 @@
...
+
+                return (
+                  <div key={index} className={this.decorateCSS("card")} style={{ backgroundColor }}>
+                    <img src={image.value} alt={title.value} className={this.decorateCSS("image")} />
+                    <div className={this.decorateCSS("text-container")}>
+                      <div className={this.decorateCSS("title")}>{title.value}</div>
+                      <div className={this.decorateCSS("subtitle")}>{subtitle.value}</div>
```
Comment (dogusmiuw) [inline editor-components/Image_gallery/image_gallery_7/image-gallery7.tsx:478]: don't render relevant elements if prop is empty
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle the changes

**Pull request closed without merging**